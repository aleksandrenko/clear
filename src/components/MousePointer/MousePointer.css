.cl-mouse-pointer {
    position: absolute;
    top: 100px;
    left: 100px;
    fill: #666;
    display: flex;
    align-content: center;
    pointer-events: none;
    z-index: 10000;
}

.cl-mouse-pointer svg {
    width: 14px;
    height: 14px;
    filter: drop-shadow(2px 2px 2px rgb(0 0 0 / 0.2));
    /*
       Compensating the offset of the svg image
    */
    margin-left: -3px;
    margin-top: -1px;
}

.cl-mouse-pointer label {
    background: #666;
    padding: 3px 6px 5px 6px;
    line-height: 1;
    font-size: x-small;
    color: #fff;
    margin-left: -4px;
    margin-top: 14px;
    box-shadow: 2px 2px 3px rgb(0, 0, 0, .2);
}
