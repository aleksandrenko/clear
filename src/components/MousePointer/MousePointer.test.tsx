import {MousePointer} from "./MousePointer";
// @ts-ignore
import * as renderer from "react-test-renderer";

it('changes the class when hovered', () => {
    const comp = <MousePointer
        x={100}
        y={200}
        label="Username"
        color="#f00"
    />;

    const component = renderer.create(comp);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
