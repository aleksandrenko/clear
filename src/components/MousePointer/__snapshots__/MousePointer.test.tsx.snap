// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<div
  className="cl-mouse-pointer"
  style={
    Object {
      "left": 100,
      "top": 200,
    }
  }
>
  <svg
    fill="#f00"
    viewBox="0 0 1000 1000"
    x="0px"
    y="0px"
  >
    <g>
      <g
        transform="translate(0.000000,512.000000) scale(0.100000,-0.100000)"
      >
        <path
          d="M1640.6,4997c-21.1-11.5-53.6-38.3-70.8-61.3c-32.5-38.3-34.5-319.7-34.5-4800.8c0-3650.4,5.7-4772.1,23-4810.4c40.2-88,178-132.1,264.2-86.1c19.1,9.6,624,652.7,1341.9,1426.1l1305.5,1406.9h1902.7c2048.2,0,1990.8-1.9,2061.6,97.6c17.2,24.9,30.6,80.4,30.6,122.5c0,76.6-5.7,82.3-3326.9,3401.5C1973.7,4859.2,1809,5020,1745.9,5020C1709.5,5020,1661.6,5008.5,1640.6,4997z"
        />
      </g>
    </g>
  </svg>
  <label
    style={
      Object {
        "backgroundColor": "#f00",
      }
    }
  >
    Username
  </label>
</div>
`;
